Tests/AsyncHTTPClientTests/ConnectionPoolTests.swift

```diff
@@ -320,6 +320,23 @@ class ConnectionPoolTests: XCTestCase {
         }
     }
 
+    func testConnectFailedWhenClosed() {
+        var state = HTTP1ConnectionProvider.ConnectionsState(eventLoop: self.eventLoop)
+        var snapshot = state.testsOnly_getInternalState()
+        snapshot.state = .closed
+        state.testsOnly_setInternalState(snapshot)
```

Comment:
can't we achieve this state by driving the state machine rather than forcing in a possibly illegal state?

Reply:
ah, good catch, we can indeed, done, thanks! (also fixed a similar test above)